pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java

```diff
@@ -323,9 +312,8 @@ public ClientBuilder clock(Clock clock) {
 
     @Override
     public ClientBuilder proxyServiceUrl(String proxyServiceUrl, ProxyProtocol proxyProtocol) {
-        if (StringUtils.isNotBlank(proxyServiceUrl) && proxyProtocol == null) {
-            throw new IllegalArgumentException("proxyProtocol must be present with proxyServiceUrl");
-        }
+        checkArgument(StringUtils.isBlank(proxyServiceUrl) || proxyProtocol != null,
+                "proxyProtocol must be present with proxyServiceUrl");
```

Comment:
Ah ?  is this equals ?

Reply:
@Technoboy- Yes, the checkArgument method's arguement should be true, so reversed

Reply:
This makes the code unreadable.

Reply:
What is the expected behavior here? Both the `proxyServiceUrl` and `proxyProtocol` can not be null or empty?

Reply:
@codelipenghui I think the expected behavior is to check that the `proxyProtocol` must be provided if the `proxyServiceUrl` is provided. I agree with @Technoboy- . I think we could make this more readable